timelock-corruption-detection/src/main/java/com/palantir/timelock/history/sqlite/SqlitePaxosStateLogHistory.java

```diff
@@ -82,16 +96,20 @@ public LearnerAndAcceptorRecords getLearnerAndAcceptorLogsSince(
         //         revisit this once we have the remote history providers set up. Also, we may have to make it
         // configurable to
         //         accommodate the rate at which logs are being published.
-        @SqlQuery("SELECT seq, val FROM paxosLog "
-                + "WHERE namespace = :namespace.value AND useCase = :useCase AND seq > :seq "
-                + "ORDER BY seq ASC LIMIT 500")
-        Map<Long, PaxosValue> getLearnerLogsSince(
-                @BindPojo("namespace") Client namespace, @Bind("useCase") String useCase, @Bind("seq") long seq);
+        @SqlQuery("SELECT seq, val FROM paxosLog WHERE namespace = :namespace.value AND useCase = :useCase AND seq >="
+                + " :lowerBound AND seq < :upperBound")
+        Map<Long, PaxosValue> getLearnerLogsInRange(
+                @BindPojo("namespace") Client namespace,
+                @Bind("useCase") String useCase,
+                @Bind("lowerBound") long lowerBound,
+                @Bind("upperBound") long upperBound);
 
-        @SqlQuery("SELECT seq, val FROM paxosLog "
-                + "WHERE namespace = :namespace.value AND useCase = :useCase AND seq > :seq "
-                + "ORDER BY seq ASC LIMIT 500")
-        Map<Long, PaxosAcceptorData> getAcceptorLogsSince(
-                @BindPojo("namespace") Client namespace, @Bind("useCase") String useCase, @Bind("seq") long seq);
+        @SqlQuery("SELECT seq, val FROM paxosLog WHERE namespace = :namespace.value AND useCase = :useCase AND seq >="
+                + " :lowerBound AND seq < :upperBound")
```

Comment:
See discussion elsewhere: I think I'd have a strong preference for having inclusive ranges here, because we've used terminology like `latestVerified` through many other parts of the code, which are not strictly correct when the upper bound has actually not been verified.